client/src/main/java/com/influxdb/client/InfluxQLQueryApi.java

```diff
@@ -92,4 +92,17 @@ InfluxQLQueryResult query(
             @Nonnull InfluxQLQuery influxQlQuery,
             @Nullable InfluxQLQueryResult.Series.ValueExtractor valueExtractor
     );
+
+    @Nonnull
+    InfluxQLQueryResult query(
+      @Nonnull InfluxQLQuery influxQlQuery,
+      @Nullable InfluxQLQuery.AcceptHeader header,
+      @Nullable InfluxQLQueryResult.Series.ValueExtractor valueExtractor
+    );
+
+    @Nonnull
+    InfluxQLQueryResult query(
+      @Nonnull InfluxQLQuery influxQLQuery,
+      @Nullable InfluxQLQuery.AcceptHeader header
+    );
```

Comment:
I think we can simplify to something like:

```suggestion
    @Nonnull
    InfluxQLQueryResult queryXYZ(@Nonnull InfluxQLQuery influxQLQuery);
```

with proper documentation detailing the differences between this and the old method.

Reply:
By `queryXYZ` I assume you mean `queryJSON` and `queryCSV` and to remove the argument `AcceptHeader header`.  A signature with the argument `valueExtractor` will still be necessary in order to parse raw results into types on the client side. This was part of the benchmarks evaluated last week.  Alternatively `valueExtractor` could be added to the `InfluxQLQuery` class.  Also a method  `query` could be used to simply opaquely implement `Accept: application/json` but the method `queryCSV` could explicitly be used to use the old method and to retain timestamps in the Epoch format.  

However I've also noticed in the "swagger" documentation (https://docs.influxdata.com/influxdb/v2/api/v1-compatibility/#operation/PostQueryV1), that two other `Accept` headers are possible, `text/csv` `application/x-msgpack`.  Since the original enhancement request only addressed `application/json` - the default, I did not explore the results of these two.  When using `AcceptHeader` as an argument in the `query` method, it occurred to me that it could be used as an extension point for covering these two other `Accept` mime types in the future.  Removing `AcceptHeader` from the `query` method signature would obscure this.

BTW msgpack has the reputation of being very fast, so perhaps these two should also be added with this PR, since this PR deals primarily with sorting out the `Accept` header of the `InfluxQLQueryService`.

Reply:
OK.  So following off/line discussion, the method names and signatures will be updated to be simplified, without `AcceptHeader header` argument.  Original methods `query` will remain but will use `Accept: application/json`.  `msgpack` and `text/csv` we'll leave out for now, until requested.